src/twisted/test/test_defer.py

```diff
@@ -11,8 +11,17 @@
 import gc, traceback
 import re
 
+from twisted.python.compat import _PY3, _PY34PLUS
+
+if _PY34PLUS:
+    from asyncio import new_event_loop, Future, CancelledError
+    asyncSkip = None
+else:
+    asyncSkip = "asyncio not available before python 3.4"
+
```

Comment:
It would be more robust here, perhaps, to catch ImportError.  As it is, I'm assuming you checked about which versions include the asyncio module, but maybe a backport of it comes along…

Anyway, seems better to test for the feature directly if you can, rather than testing for interpreter versions, and here you should be able to test for the feature.